src/ripple/app/paths/FindPaths.cpp

```diff
@@ -90,24 +94,30 @@ FindPaths::FindPaths (
     AccountID const& srcAccount,
     AccountID const& dstAccount,
     STAmount const& dstAmount,
+    boost::optional<STAmount> const& srcAmount,
     int level,
     unsigned int maxPaths)
         : impl_ (std::make_unique<Impl> (
-              cache, srcAccount, dstAccount, dstAmount, level, maxPaths))
+            cache, srcAccount, dstAccount,
+                dstAmount, srcAmount, level, maxPaths))
 {
 }
 
 FindPaths::~FindPaths() = default;
 
-bool FindPaths::findPathsForIssue (
+boost::optional<std::pair<
+    STPathSet, std::vector<STAmount>>>
+FindPaths::findPathsForIssue (
     Issue const& issue,
-    STPathSet& pathsInOut,
+    STPathSet const& paths,
     STPath& fullLiquidityPath)
 {
-    return impl_->findPathsForIssue (issue, pathsInOut, fullLiquidityPath);
+    return impl_->findPathsForIssue (issue, paths, fullLiquidityPath);
 }
 
-bool findPathsForOneIssuer (
+boost::optional<std::pair<
+    STPathSet, std::vector<STAmount>>>
```

Comment:
I think this pair `std::pair<STPathSet, std::vector<STAmount>>` is important enough to have its own struct - particularly since I so far haven't quite figured out what it is.  :-)